test/utils/Partition_One_Test.cpp

```diff
@@ -0,0 +1,147 @@
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include <stdio.h>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/lexical_cast.hpp>
+#include <vector>
+
+//This way we can test the partition_one, since this doesn't have an explicit MPI dependency
+#define NGEN_MPI_ACTIVE
+#include "core/Partition_One.hpp"
+#include "FileChecker.h"
+
+
+class PartitionOneTest: public ::testing::Test {
+
+    protected:
+
+    std::vector<std::string> hydro_fabric_paths;
+
+    std::string catchmentDataFile;
+    geojson::GeoJSON catchment_collection;
+
+    std::vector<std::string> catchment_subset_ids;
+    std::vector<std::string> nexus_subset_ids;
+
+    std::unordered_set<std::string> catchment_ids;
+    std::unordered_set<std::string> nexus_ids;
+
+    Partition_One partition_one;
+    PartitionData partition_data;
+
+    PartitionOneTest() {} 
+
+    ~PartitionOneTest() override {}
+
+    std::string file_search(const std::vector<std::string> &parent_dir_options, const std::string& file_basename)
+    {
+        // Build vector of names by building combinations of the path and basename options
+        std::vector<std::string> name_combinations;
+
+        // Build so that all path names are tried for given basename before trying a different basename option
+        for (auto & path_option : parent_dir_options)
+            name_combinations.push_back(path_option + file_basename);
+
+        return utils::FileChecker::find_first_readable(name_combinations);
+    }
+
+    void read_file_generate_partition_data()
+    {
+        const std::string file_path = file_search(hydro_fabric_paths, "catchment_data.geojson");
+        catchment_collection = geojson::read(file_path, catchment_subset_ids);
+
+        for(auto& feature: *catchment_collection)
+        {
+            std::string cat_id = feature->get_id();
+            partition_data.catchment_ids.emplace(cat_id);
+            std::string nex_id = feature->get_property("toid").as_string();
+            partition_data.nexus_ids.emplace(nex_id);
+        }
+    }
+
+    void SetUp() override;
+
+    void TearDown() override;
+
+    void setupArbitraryExampleCase();
+
+};
+
+void PartitionOneTest::SetUp() {
+    setupArbitraryExampleCase();
+}
+
+void PartitionOneTest::TearDown() {
+
+}
+
+void PartitionOneTest::setupArbitraryExampleCase() {
+    hydro_fabric_paths = {
+        "data/",
+        "./data/",
+        "../data/",
+        "../../data/",
+
+    };
+}
+
+TEST_F(PartitionOneTest, TestPartitionData_1a)
+{
+    read_file_generate_partition_data();
+
+    partition_one.generate_partition(catchment_collection);
+    PartitionData data_struct = partition_one.partition_data;
+    catchment_ids = data_struct.catchment_ids;
```

Comment:
I just realized an oversight in these tests - we also need to check that `catchment_ids` and `nexus_ids` contain the right number of elements.

Also, I just noticed that the comparison isn't the partition output against the original `catchment_collection`, but just against itself. The test as a whole should affirm that the set in the single partition is identical to the set in the input it was given.